src/test/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/HearingBundleReadyNotificationHandlerTest.java

```diff
@@ -0,0 +1,202 @@
+
+@ExtendWith(MockitoExtension.class)
+@SuppressWarnings("unchecked")
+class HearingBundleReadyNotificationHandlerTest extends AbstractNotificationsHandlerTestBase {
+
+    @Mock
+    private HomeOfficeInstructService homeOfficeInstructService;
+    @Mock
+    protected ListingNotificationHelper listingNotificationHelper;
+
+    @Captor
+    private ArgumentCaptor<HearingInstructMessage> hearingInstructMessageArgumentCaptor;
+
+    private HearingBundleReadyNotificationHandler hearingBundleReadyNotificationHandler;
+
+    Hearing hearing = Hearing.HearingBuilder.hearing()
+        .withHearingType("ORAL")
+        .withHmctsHearingRef("ariaListingReference")
+        .build();
+
+    HearingInstructMessage hearingInstructMessage =
+        HearingInstructMessage.HearingInstructMessageBuilder
+            .hearingInstructMessage()
+            .withConsumerReference(consumerReference)
+            .withHoReference(someDocumentReference)
+            .withMessageHeader(messageHeader)
+            .withMessageType(HEARING_BUNDLE_READY.name())
+            .withHearing(hearing).build();
+
+    @BeforeEach
+    void setUp() {
+
+        hearingBundleReadyNotificationHandler =
+            new HearingBundleReadyNotificationHandler(
+                homeOfficeInstructService,
+                notificationsHelper,
+                listingNotificationHelper);
+    }
+
+    @ParameterizedTest
+    @EnumSource(value = Event.class, names = {"GENERATE_HEARING_BUNDLE", "CUSTOMISE_HEARING_BUNDLE"})
+    @MockitoSettings(strictness = Strictness.WARN)
+    void check_handler_returns_case_data_for_valid_input(Event event) {
+
+        setupCase(event);
+        setupCaseData();
+        setupHelperResponses();
+        when(homeOfficeInstructService.sendNotification(any())).thenReturn("OK");
+
+        when(listingNotificationHelper.getHearingBundleReadyInstructMessage(
+                any(AsylumCase.class), any(),
+                any(), anyString())).thenReturn(hearingInstructMessage);
+
+        PreSubmitCallbackResponse<AsylumCase> response =
+            hearingBundleReadyNotificationHandler.handle(ABOUT_TO_SUBMIT, callback);
+
+        assertThat(response).isNotNull();
+        assertThat(response.getData()).isNotEmpty();
+        assertThat(response.getData()).isEqualTo(asylumCase);
+        assertTrue(response.getErrors().isEmpty());
+        verify(asylumCase, times(1)).write(HOME_OFFICE_HEARING_BUNDLE_READY_INSTRUCT_STATUS, "OK");
+        verify(homeOfficeInstructService).sendNotification(hearingInstructMessageArgumentCaptor.capture());
+
+        final HearingInstructMessage instructMessage = hearingInstructMessageArgumentCaptor.getValue();
+        assertNotificationInstructMessage(instructMessage);
+    }
+
+    @ParameterizedTest
+    @EnumSource(value = Event.class, names = {"GENERATE_HEARING_BUNDLE", "CUSTOMISE_HEARING_BUNDLE"})
+    @MockitoSettings(strictness = Strictness.WARN)
```

Comment:
Have you verified if it is really needed here? I remeber it was used initially to overcome an issue. I do not want to just have it to replicate the other tests. Could you please verify it once?

Reply:
Removed settings and corrected